contracts/wallet.sol

```diff
@@ -33,16 +33,68 @@ import "./externals/ERC165.sol";
 import "./externals/ECDSA.sol";
 
 
-/// @title ControllableOwnable combines Controllable and Ownable
-/// @dev providing an additional modifier to check if Owner or Controller
-contract ControllableOwnable is Controllable, Ownable {
-    /// @dev Check if the sender is the Owner or one of the Controllers
-    modifier onlyOwnerOrController() {
-        require (_isOwner(msg.sender) || _isController(msg.sender), "only owner||controller");
+/// @title OptOutableMonolith2FA is used a configurable 2FA.
+/// @dev This provides the various modifiers and utility functions needed for 2FA.
+/// @dev 2FA is needed to confirm changes to the security settings in the wallet.
+contract OptOutableMonolith2FA is Controllable, Ownable {
+
+    event SetMonolith2FA(address _sender); 
+    event SetPersonal2FA(address _sender, address _p2FA);
+
+    bool public monolith2FA;
+    address public personal2FA;
+
+    constructor() public {
+        monolith2FA = true;
+    }
+
+    // @dev This modifier ensures that a method is only accessible to 2nd factor
+    modifier only2FA() {
+        if (monolith2FA) {
+            require(_isController(msg.sender), "sender is not a controller");
+        } else {
+            require(msg.sender == personal2FA, "sender is not personal 2FA");
+        }
+        _;
+    }
+
+    /// @dev Check if the sender is the Owner or 2FA
+    modifier onlyOwnerOr2FA() {
+        require (_isOwner(msg.sender) || _is2FA(msg.sender), "only owner or 2FA");
         _;
     }
+
+    /// @dev set Monolith to be the 2FA
+    function setMonolith2FA() external onlyOwner {
+        require(!monolith2FA, "monolith2FA enabled");
+
+        monolith2FA = true;
+        personal2FA = address(0);
+
+        emit SetMonolith2FA(msg.sender);
+    }
+
+    /// @dev set personal 2FA to the address the user provided
+    function setPersonal2FA(address _p2FA) external onlyOwner {
+        require(_p2FA != address(0), "2FA cannot be set to zero");
+
+        personal2FA = _p2FA;
+        monolith2FA = false;
```

Comment:
Would it make sense to allow both personal and monolith 2fa to be enabled at the same time instead of only allowing one of them to be enabled? This would mean that the user could just add a personal 2FA without removing monolith 2FA functionality.